### PR TITLE
Draft: local metrics subpackage

### PR DESCRIPTION
**Pull request by Netcodez**
Code Review is needed. Code working as expected and saves the h5 files.
Need an efficient way to load files onto the metric_calc  function when calling it in the AE and SE module.

### CODE REVIEW

File: cometr/local_metrics/AE.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import h5py
+import numpy as np
+import torch
+from beartype import beartype
+
+
+from cometr.Metric import Metric
+
+
+class AE(Metric):
```
Comment (nickcrawford1) [inline cometr/local_metrics/AE.py:14]: perhaps consider a more descriptive class name to allow the class to 'self document' when used.

for example, using this class currently, would look something like this

```
from cometr.local_metrics.AE import AE
object = AE(metric)
```

whereas if you name the classes and functions clearly, the user can abbreviate them when they want to shorten but when just glancing at code they right, you can quickly identify what its doing. example

```
from cometr.local_metrics.AbsoluteError import AbsoluteErrorCalculation
object = AbsoluteErrorCalculation(metric)
```

This is all preference and hard to show in an example but imagine you have 300 lines of code between your import and object. If the object is running AE, you have to go search for what that means whereas with my suggestion its clear what that object does.

As i said, this is preference but to make the package clearer, this might be nice

Comment (Netcodez) [reply cometr/local_metrics/AE.py]: Thanks. I have updated it to be more descriptive.

**Review by LAShemilt — COMMENTED** [review]
Looks fine to me, happy for merge. For future it might be wise to split these up into separate PRs, one for code + tests ,  and one for docs.

One concern in `tests/data/output.txt` the number has changed quite a lot- why is this?

**Comment by nickcrawford1** [comment]
I agree with Laura and this is where planning the work out and splitting up the tasks is vert important and allows a reviewer to focus on a specific element. I would expand and say in the future, think about the tasks you've been set and if they all need to be in the same PR or if you can work on one on a branch and the other on a different branch. 

**example:**

- Absolute error could have been written in a branch
- a smaller pull request for absolute error could have been launched early
- then to do the squared error you could have:
    - branched off the absolute error branch (if you needed that code
or 
    - just created a new branch from master to work on the squared error separately.
- Then also have a branch running from master called tests or something and have those separate again
- finally also have a branch for documentation to keep it all separate.


If you work something like this, it will:

- makes the PR's smaller and more likely to get a review quickly, 
- allows you to work on other bits while waiting for PR's to be reviewed
- makes the changes clearer and focuses the work.

**To do this well you can use the local git merge method I showed you the other day to keep the branch you're working on up to date, and you should see the PR process speed up**

The end result would have been 4 easy to read/see focused PR's, created throughout the build that can be reviewed while you work on other things. They are also more likely to get a review because the look less imposing than one massive one with lots of files changed and lost of different types of work clustered into one giant request.

As I said, this is just a suggestion to think about as a way of working. It work for me but you'll likely find your own method as you gain experience. Any questions, feel free to ask.

**Comment by Netcodez** [comment]
> Looks fine to me, happy for merge. For future it might be wise to split these up into separate PRs, one for code + tests , and one for docs.
> 
> One concern in `tests/data/output.txt` the number has changed quite a lot- why is this?
Thank you for the feedback.  I will follow suit for future pull requests. 
The output.txt file contains the result of any metric that is run. That is why it changes according to the metric.  However, I have refactored the code so the name of the .txt file is the same name as the test run for easier identification.

**Comment by Netcodez** [comment]
> I agree with Laura and this is where planning the work out and splitting up the tasks is vert important and allows a reviewer to focus on a specific element. I would expand and say in the future, think about the tasks you've been set and if they all need to be in the same PR or if you can work on one on a branch and the other on a different branch.
> 
> **example:**
> 
> * Absolute error could have been written in a branch
> * a smaller pull request for absolute error could have been launched early
> * then to do the squared error you could have:
>   
>   * branched off the absolute error branch (if you needed that code
>     or
>   * just created a new branch from master to work on the squared error separately.
> * Then also have a branch running from master called tests or something and have those separate again
> * finally also have a branch for documentation to keep it all separate.
> 
> If you work something like this, it will:
> 
> * makes the PR's smaller and more likely to get a review quickly,
> * allows you to work on other bits while waiting for PR's to be reviewed
> * makes the changes clearer and focuses the work.
> 
> **To do this well you can use the local git merge method I showed you the other day to keep the branch you're working on up to date, and you should see the PR process speed up**
> 
> The end result would have been 4 easy to read/see focused PR's, created throughout the build that can be reviewed while you work on other things. They are also more likely to get a review because the look less imposing than one massive one with lots of files changed and lost of different types of work clustered into one giant request.
> 
> As I said, this is just a suggestion to think about as a way of working. It work for me but you'll likely find your own method as you gain experience. Any questions, feel free to ask.

Thanks for the review. It is definitely something I need to implement.
